src/clients/advanced/TokenLifespan.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { Control, Controller } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import {
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Split,
+  SplitItem,
+} from "@patternfly/react-core";
+
+import {
+  TimeSelector,
+  Unit,
+} from "../../components/time-selector/TimeSelector";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+
+type TokenLifespanProps = {
+  id: string;
+  name: string;
+  defaultValue: string;
+  control: Control<Record<string, any>>;
+  units?: Unit[];
```

Comment:
The `units` prop is always specified, so I think it can be made required.
```suggestion
  units: Unit[];
```

Reply:
but it's optional for the `TimeSelector`

Reply:
Sure, but all components that use this one provide them. IMHO it's best to be as strict as possible out of the box, you can always loosen the type when needed. This prevents logic from being introduced that handles possible empty values, when if fact this logic is redundant.

Reply:
if you leave it empty you get all the units `["seconds", "minutes", "hours", "days"]`

Reply:
Sure, but if someone ever makes logic inside of this component and touches these props they have to take into account it could be empty.